### PR TITLE
Refactoring replication

### PR DESCRIPTION
**Pull request by kosssi**

### CODE REVIEW

File: www-src/app/replicator/replicator_migration.coffee [hunk 1]
```diff
@@ -1,4 +1,5 @@
 async = require 'async'
+PouchDB = require 'pouchdb'
```
Comment (nono) [inline www-src/app/replicator/replicator_migration.coffee]: Why PouchDB is required in this file?

File: www-src/app/replicator/replicator_config.coffee [hunk 2]
```diff
@@ -37,14 +50,21 @@ module.exports = class ReplicatorConfig extends Backbone.Model
                 @remote = @createRemotePouchInstance()
                 callback? null, this
 
-    getScheme: () ->
+        options = @makeDSUrl('/filters/config')
+        options.body = @getConfigFilter()
+        request.put options, (err, res, body) =>
+            return callback err if err
+            return callback body unless body.success or body._id
+            callback? null, this
```
Comment (nono) [inline www-src/app/replicator/replicator_config.coffee]: It's weird that `callback` is mandatory on the first two cases but optional in the third.

File: www-src/app/replicator/main.coffee [hunk 3]
```diff
@@ -1,4 +1,5 @@
 async = require 'async'
+PouchDB = require 'pouchdb'
```
Comment (nono) [inline www-src/app/replicator/main.coffee]: Why PouchDB is required here?

File: www-src/app/replicator/replicator_config.coffee [hunk 4]
```diff
@@ -15,6 +17,17 @@ module.exports = class ReplicatorConfig extends Backbone.Model
...
+    getConfigFilter: ->
+        compare = "doc.docType === 'file' or doc.docType === 'folder'"
+        compare += " or doc.docType === 'contact'" if @get "syncContacts"
+        compare += " or doc.docType === 'event'" if @get "syncCalendars"
+        if @get "cozyNotifications"
+            compare += " or (doc.docType === 'notification'"
+            compare += " and doc.type === 'temporary')"
+
+        'filters':
+            'config': "function (doc) { return #{compare} }"
```
Comment (nono) [inline www-src/app/replicator/replicator_config.coffee]: The single quotes are optional here, and I think we should omit them.

File: www-src/app/replicator/replicator_migration.coffee (hunk 1 above)
Comment (kosssi) [inline www-src/app/replicator/replicator_migration.coffee]: I'm changing import of CouchDB.

Now it's only with require ;)

File: www-src/app/replicator/main.coffee (hunk 3 above)
Comment (kosssi) [inline www-src/app/replicator/main.coffee]: I'm changing import of CouchDB.

Now it's only with require ;)

**Pull request merged**